spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark
+
+// scalastyle:off import.ordering.noEmptyLine
+import java.util.Locale
+
+import scala.util.Try
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.network.util.{ByteUnit, JavaUtils}
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+
+trait DeltaSharingOptionParser {
```

Comment:
This seems a bit odd. The parser trait does not seem to do any parsing. 
Do you plan to add some parsing utilities here?
Should we call this DeltaSharingOptionsBase instead?

Reply:
The `toBoolean` function actually parses the input parameter value, no plan to add other parsing utilities. I feel that's why it's called parser.